src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs

```diff
@@ -87,6 +93,8 @@ protected override void OnEventCommand(EventCommandEventArgs command)
                 _ilBytesJittedCounter ??= new PollingCounter("il-bytes-jitted", this, () => System.Runtime.JitInfo.GetCompiledILBytes()) { DisplayName = "IL Bytes Jitted", DisplayUnits = "B" };
                 _methodsJittedCounter ??= new PollingCounter("methods-jitted-count", this, () => System.Runtime.JitInfo.GetCompiledMethodCount()) { DisplayName = "Number of Methods Jitted" };
                 _jitTimeCounter ??= new IncrementingPollingCounter("time-in-jit", this, () => System.Runtime.JitInfo.GetCompilationTime().TotalMilliseconds) { DisplayName = "Time spent in JIT", DisplayUnits = "ms", DisplayRateTimeScale = new TimeSpan(0, 0, 1) };
+
+                AppContext.LogSwitchValues(this);
```

Comment:
The more traditional pattern is to store a static instance of the source

```csharp
public static RuntimeEventSource Log = new RuntimeEventSource();
```

and then access that anywhere you use it. So, the inverse of the access as written.

I think that means you'd need to change the visibility of the `s_SuntimeEventSource` field and potentially remove the `Initialize` method in exchange for the static initializer.

Reply:
I took a look at where it's initialized (`System.StartupHookProvider.ProcessStartupHooks`). I don't think there would be any side-effects related to removing the `Initialize` method and relying on the static initializer. I'm okay leaving it as is, though.